sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/BinaryData.java

```diff
@@ -276,10 +295,63 @@ public String toString(Charset charSet) {
      * @param clazz representing the type of the Object.
      * @param serializer to use deserialize data into type.
      * @param <T> Generic type that the data is deserialized into.
-     * @throws NullPointerException if {@code clazz} or {@code serializer} is null.
+     * @throws NullPointerException If {@code clazz} or {@code serializer} is null.
      * @return The {@link Object} of given type after deserializing the bytes.
      */
     public  <T> Mono<T> toObjectAsync(Class<T> clazz, ObjectSerializer serializer) {
+
+        if (Objects.isNull(serializer)) {
+            return monoError(LOGGER, new NullPointerException("'serializer' cannot be null."));
```

Comment:
The other `toObject()` overload also checks if the `clazz` is null. We should add that check here as well.